### PR TITLE
Sinatra 2.0.0 alpha [redux]

### PR DESCRIPTION
**Pull request by TrevorBramble**
(Reopening #1029 as a branch on core project to allow collaborator activity)
(Holy crap I named the branch with "2.2.0" but I'm not switching it around again. Sorry!)

Here we go!

The Plan:
 - Drop support for Rubies earlier than 2.2 (MRI and equivalent implementations)
 - Change nothing else
 - Move 1.4 development to the `1.4.x` branch, where releases will come from until 2.0 is fully released

Tasks:
 - [X] Revise Travis matrix
 - [X] Update gem version
 - [X] Add Ruby version dependency to gemspec
 - [x] Raise Rack version requirement to `~> 2.0`, fix new incompatibilities
 - [ ] Remove overt conditional statements for dropped Rubies
 - [ ] Update English README to communicate new set of supported versions related notes
 - [ ] Coordinate translations for non-English READMEs
 - [ ] Sweep outstanding issues for 1.4.x changes before shunting to maintenance branch
 - [ ] ...figure out what else should be done...

### CODE REVIEW

**Comment by burningTyger** [comment]
Looking at the Gemfile I see all the dependencies on templates. Would it be a good idea to drop all dependencies and let the user require them?

File: README.md [hunk 1]
```diff
@@ -2871,31 +2871,10 @@ thin --threaded start
...
-    1.9.3 is fully supported and recommended. Please note that switching to 1.9.3
-    from an earlier version will invalidate all sessions. 1.9.3 will be supported
-    until the release of Sinatra 2.0.
-  </dd>
-
-  <dt>Ruby 2.x</dt>
-  <dd>
-    2.x is fully supported and recommended. There are currently no plans to drop
-    official support for it.
+    2.2 is fully supported and recommended. There are currently no plans to
```
Comment (burningTyger) [inline README.md]: why not support the ruby version available on the release date? i.e. 2.3 if ruby has had a release before sinatra 2.0.

**Comment by beanieboi** [comment]
hey, i would love to help.

it feels like the TODO list is a little outdated, would you mind updating the TODO list so people can pickup tasks?

thanks a bunch!
ben

**Comment by ericboehs** [comment]
Would it make sense to cut an alpha release on rubygems?

**Comment by b264** [comment]
https://github.com/sinatra/sinatra/issues/1123

**Pull request merged**